### ISSUE TITLE
PythonAlgoEditDialog sometimes appears completely off-screen

### ISSUE THREAD

**Issue by agilmore2** [opendcs/opendcs#61]
We have one user who has some lengthy Python algorithms, and when they open the Python editor, the dialog does not show on screen, and compedit appears completely frozen.

An examination of the .loc files show that the dialog writes a huge negative y value for position. Unfortunately, no methods work to move the editor back on screen. Dialogs are modal and do not show up in the alt-tab list, cannot have a "Move" option enabled, do not obey Windows logo key+arrow commands.

This only happens for this user, other users can open the same code just fine. The same user can open other Python algorithms just fine.

If the amount of code in the problematic algorithm is greatly decreased by removing blocks from the cp_algo_script table, the user is able to open it again.

This happens on a hosted VM near the database, and on the users laptop. It did not happen to them when using the GUIs from Linux. OS is probably Windows 10.

Mike M has a copy of the problematic algorithm XML.

Changing the appropriate .loc file to read-only after viewing a "good" dialog did not help. Turning off the option to remember screen locations also did not help.

Currently having them test a kludgy fix to PythonAlgoEditDialog.java:

```
@OverRide
public void trackChanges(String frameTitle) {
super.trackChanges(frameTitle);

	super.setLocation(super.getX(), Math.max(super.getY(), 0));
}
```

Any ideas?

**Comment by agilmore2** [comment]
Very likely fixed by #35, as the modal dialog was freezing the editor as it appeared offscreen.

Fix in the PR is much more general though, much appreciated!

Will promote 7.0 to the user for testing of this fix!

**Comment by agilmore2** [comment]
GUI updates in 7.0 fixed this!
